test/blst/aggregateWithRandomness.test.ts

```diff
@@ -0,0 +1,183 @@
+import {beforeAll, describe, expect, it} from "bun:test";
+import {
+	PublicKey,
+	Signature,
+	aggregatePublicKeys,
+	aggregateSerializedSignatures,
+	aggregateWithRandomness,
+	asyncAggregateWithRandomness,
+	verifyMultipleAggregateSignatures,
+} from "../../src/blst.ts";
+import {G1_POINT_AT_INFINITY, G2_POINT_AT_INFINITY} from "./__fixtures__/index.js";
+import {expectNotEqualHex, getTestSet, getTestSetsSameMessage} from "./utils/index.js";
+
+describe("Aggregate With Randomness", () => {
+	const sameMessageSets = getTestSetsSameMessage(10);
+	const msg = sameMessageSets.msg;
+	const sets = sameMessageSets.sets.map((s) => ({
+		msg: msg,
+		pk: s.pk,
+		sig: s.sig.toBytes(),
+	}));
+	const randomSet = getTestSet(20);
+	const infinityPublicKey = Buffer.from(G1_POINT_AT_INFINITY, "hex");
+
+	beforeAll(() => {
+		// make sure sets are valid before starting
+		expect(() => PublicKey.fromBytes(infinityPublicKey).keyValidate()).toThrow("PkIsInfinity");
+		const sig = Signature.fromBytes(sets[0].sig);
+		expect(sig.verify(msg, sets[0].pk)).toBeTrue();
+
+		expectNotEqualHex(msg, randomSet.msg);
+		expect(randomSet.sig.verify(randomSet.msg, randomSet.pk)).toBeTrue();
+		expect(verifyMultipleAggregateSignatures([randomSet])).toBeTrue();
+	});
+
+	describe("aggregateWithRandomness()", () => {
+		it("should not accept an empty array argument", () => {
+			expect(() => aggregateWithRandomness([])).toThrow("At least one PkAndSerializedSig is required");
+		});
+		it("should throw for invalid serialized", () => {
+			expect(() =>
+				aggregateWithRandomness(
+					sets.concat({
+						pk: sets[0].pk,
+						//TODO: (@matthewkeil) this throws error "Public key is infinity" not signature because there is only one blst error
```

Comment:
what does this TODO mean, is this still relevant? @matthewkeil

Reply:
So it shouldn't be a TODO, but a note?

Reply:
yeah